src/utils/core.js

```diff
@@ -102,14 +105,35 @@ const validateGroups = (req, groups, all) => {
 /*
  * Authentication middleware wrapper
  */
-module.exports.isAuthenticated = (groups = [], all = false) => (req, res, next) => {
-  if (req.session.user && validateGroups(req, groups, all)) {
-    return next();
+module.exports.isAuthenticated = (core, groups = [], all = false) => (req, res, next) => {
```

Comment:
This now has side-effects and does not do `isAuthenticated` anymore, but some extras.

Split out your token related checking and refreshing into a new middleware called something like `useWebTokens` that you can register before the `isAuthenticated`.